### PR TITLE
Add support for Tz operator (horizontal scaling).

### PR DESCRIPTION
**Pull request by msbit**
Simple implementation of the Tz operator for allowing horizontal scaling of text.

### CODE REVIEW

**Comment by pointlessone** [comment]
Hello @msbit. Thank you very much for your contribution.

I'll take a closer look at the code a bit later. Meanwhile could you please add a spec or two that cover new code?

**Comment by msbit** [comment]
Hey @pointlessone. I've based the implementation on character_spacing, and so I was thinking of adding them into a PR for https://github.com/prawnpdf/prawn, as all the character_spacing specs seem to be in place on there. Would that be okay?

**Comment by pointlessone** [comment]
This is unfortunate consequence of extraction of PDF Core out of Prawn. There isn't always an obvious way to cut it.

Those specs definitely should belong here since the code is in this gem. Could you please file an issue about that since you've already inspected the code?

As for this PR we'd prefer specs here for this code and in Prawn gem for the code that uses this feature if any.

**Comment by msbit** [comment]
Okay, that sounds fair.

I'll look at the specs for this PR tomorrow, and while I'm investigating that I'll see if I can put together a reasonable overview of what would need to come across in terms of specs from prawn into pdf-core for character_spacing.

**Comment by pointlessone** [comment]
@msbit Thank you. Much appreciated.

**Comment by msbit** [comment]
Added specs covering the pdf-core specific parts of this.

File: spec/text_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+      attr_reader :text
+
+      def add_content(str)
+        @text ||= ""
+        @text << str
+      end
+    end
+    @mock = TextMock.new
+  end
```
Comment (pointlessone) [inline spec/text_spec.rb]: It probably would be better to move class definition out of `before` and use `let` instead to create a new mock for each example.

File: spec/text_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  describe "horizontal_text_scaling" do
+    describe "called without argument" do
+      before do
+        @result = @mock.horizontal_text_scaling
```
Comment (pointlessone) [inline spec/text_spec.rb]: This line can be moved into the next example. Generally it's recommended to use `let` syntax or straight inline setup instead of instance variables.

File: lib/pdf/core.rb [hunk 3]
```diff
@@ -16,6 +16,7 @@
 require_relative "core/outline_root"
 require_relative "core/outline_item"
 require_relative "core/renderer"
+require_relative "core/text"
```
Comment (pointlessone) [inline lib/pdf/core.rb]: Why did you add this require?

Comment (msbit) [inline lib/pdf/core.rb]: That was to bring in the PDF::Core::Text class for the specs, the other option is to require it from the spec, or spec_helper, but it seemed odd that it wouldn't be required in the first place.

File: spec/text_spec.rb (hunk 1 above)
Comment (msbit) [inline spec/text_spec.rb]: Yep, that makes sense.

File: spec/text_spec.rb (hunk 2 above)
Comment (msbit) [inline spec/text_spec.rb]: Yep, wrapping that in a let, as above. Shouldn't be too tricky.

**Comment by msbit** [comment]
👌
